### ISSUE TITLE
MD053 false positive when link contains code

### ISSUE THREAD

**Issue by davidhewitt** [rvben/rumdl#128]
With the following markdown text

```markdown
See [`Bound<'_, PyAny>::is_callable`] function.

[`Bound<'_, PyAny>::is_callable`]: foo
```

I get a false positive

```
[MD053] Unused link/image reference: [`bound<'_, pyany>::is_callable`]
```

I think this is the last false positive I've seen inside PyO3 - thanks for making so many improvements on our behalf!

**Comment by rvben** [comment]
Hi @davidhewitt,

Thanks for raising the issue! It's nice to have rich examples from the PyO3 repository, because these things are hard to predict without having some real life scenario's. Glad to hear this is one of the last false positives! 

Thanks again. Your issues have really helped me improve rumdl for everyone. I expect the fix for this to land in the next release in the coming days.

**Comment by rvben** [comment]
The fix landed in 0.0.170. 🚀

**Comment by davidhewitt** [comment]
Awesome, many thanks! I just tested, and the example in the OP is now fixed however I hit two more which still trigger the false positive:

```markdown
[`CompareOp::matches`]: ...
[`inspect.signature`]: ...
```
